libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -4310,41 +4247,42 @@ bool WaveChannel::ClipsAreUnique() const
    return intervals.size() == Set{ begin, end }.size();
 }
 
-WaveTrack &WaveChannel::GetNarrowTrack()
-{
-   auto index = GetChannelIndex();
-   auto iter = TrackList::Channels(&GetTrack()).begin();
-   std::advance(iter, index);
-   return **iter;
-}
-
-const WaveTrack &WaveChannel::GetNarrowTrack() const
-{
-   return const_cast<WaveChannel&>(*this).GetNarrowTrack();
-}
-
 WaveClipHolders &WaveChannel::Clips()
 {
-   return GetNarrowTrack().mClips;
+   return mClips;
 }
 
 const WaveClipHolders &WaveChannel::Clips() const
 {
    return const_cast<WaveChannel&>(*this).Clips();
 }
 
-void WaveTrack::ZipClips(bool really)
+void WaveTrack::ZipClips()
 {
    const auto pOwner = GetOwner();
-   assert(GetOwner() && TrackList::Channels(this).size() == 2);
-
-   // TODO wide wave tracks -- really!
-   if (!really)
+   assert(GetOwner()); // pre
+   assert(NChannels() == 1); // pre
+
+   // If deserializing, first un-link the track, so iterator finds the partner.
+   this->SetLinkType(LinkType::None);
+   auto iter = pOwner->Find(this);
+   assert(this == *iter);
+   ++iter;
+   assert(iter != pOwner->end()); // pre
+   auto pRight = dynamic_cast<WaveTrack*>(*iter);
+   assert(pRight && pRight->NChannels() == 1); // pre
+
+   //! Refuse if clips are not well aligned.
+   if (!AreAligned(this->SortedClipArray(), pRight->SortedClipArray()))
       return;
```

Comment:
Shouldn't the modification `this->SetLinkType(LinkType::None);` be undone here?

Reply:
Precisely not!

The tentative stereo channel pair (as indicated by the "leader" attribute) is discovered to be ineligible to merge and so two mono tracks remain.